src/main/java/com/box/sdk/BoxAPIRequest.java

```diff
@@ -83,6 +89,39 @@ public BoxAPIRequest(BoxAPIConnection api, URL url, String method) {
 
         this.addHeader("Accept-Encoding", "gzip");
         this.addHeader("Accept-Charset", "utf-8");
+
+        // Setup the SSL context manually to force newer TLS version on legacy Java environments
+        // This is necessary because Java 7 uses TLSv1.0 by default, but the Box API will need
+        // to deprecate this protocol in the future.  To prevent clients from breaking, we must
+        // ensure that they are using TLSv1.1 or greater!
+        SSLContext sc = null;
+        try {
+            // Start with the default version for forwards compatibility with TLSv1.3 and later
+            sc = SSLContext.getDefault();
+            if (!sc.getProtocol().equals("TLSv1")) {
+                // Try to upgrade to a higher TLS version
+                sc = SSLContext.getInstance("TLSv1.1");
+                sc = SSLContext.getInstance("TLSv1.2");
+            }
+        } catch (NoSuchAlgorithmException ex) {
+            if (sc == null) {
+                LOGGER.warning("Unable to set up SSL context for HTTPS!  This may result in the inability "
+                        + " to connect to the Box API.");
+            }
+            if (sc == null || sc.getProtocol().equals("TLSv1")) {
```

Comment:
There's a `sc == null` check above which means two warnings will be posted

Reply:
Good catch!  I'll dedupe these.